Library/ViewModels/ShippingRulesViewModel.swift

```diff
@@ -38,10 +39,18 @@ public final class ShippingRulesViewModel: ShippingRulesViewModelType,
     .map(second)
     .skipNil()
 
-    let dataSelected = Signal.combineLatest(
+    let selectedIndex = Signal.combineLatest(
       dataInitial,
       self.didSelectShippingRuleAtIndexProperty.signal.skipNil()
     )
+    .map { data, newIndex in (data.1.firstIndex(of: data.2), newIndex) }
```

Comment:
I wonder if the would be more readable to unpack `dataInitial` by doing 
```
let selectedIndex = Signal.combineLatest(
      dataInitial.map(unpack),
      self.didSelectShippingRuleAtIndexProperty.signal.skipNil()	      
    )
```

And use explicit names here
```
.map { data, newIndex in (data.1.firstIndex(of: data.2), newIndex) }
```

So it would become
```
.map { _, shippingRules, selectedRule, newIndex in 
    (shippingRules.firstIndex(of: selectedRule), newIndex) 
}
```

Reply:
Good idea...I'd love to do that but it won't compile ... not really sure what's wrong.

<img width="1092" alt="Screen Shot 2019-09-03 at 4 18 15 PM" src="https://user-images.githubusercontent.com/387596/64214753-7d8eb000-ce66-11e9-8874-d446f34df91b.png">

Reply:
Yeah, the compiler must be confused due some type inference. I'm ok to left as it is then, unless you think it's worth it investigating a little more.